Classes/Search.php

```diff
@@ -312,15 +312,88 @@ public function getResponseBody()
         return $this->getResponse()->response;
     }
 
+	/**
+	 * Returns a list of solr fields that could be retrieved without applying htmlspecialchars
+	 *
+	 * @return array
+	 */
+	protected function getSkipEscapeSolrFields()
+	{
+		$trustedFieldArray = array();
+		if(!isset($GLOBALS['TSFE']->tmpl->setup['plugin.']['tx_solr.']['search.']['skipEscapeFields']) ||
+			!is_string($GLOBALS['TSFE']->tmpl->setup['plugin.']['tx_solr.']['search.']['skipEscapeFields'])) {
+			return $trustedFieldArray;
+		}
+
+		$trustedFields = $GLOBALS['TSFE']->tmpl->setup['plugin.']['tx_solr.']['search.']['skipEscapeFields'];
+		$trustedFieldArray = GeneralUtility::trimExplode(",", $trustedFields);
+
+		return $trustedFieldArray;
+	}
+
     /**
-     * Gets the result's documents
-     *
-     * @return \Apache_Solr_Document[]
+     * @deprecated use getResultEscapedDocuments now when possible
+     * @return array
      */
-    public function getResultDocuments()
-    {
-        return $this->getResponseBody()->docs;
-    }
+	public function getResultDocuments()
+	{
+		return $this->getResponseBody()->docs;
+	}
+
+	/**
+	 * Returns all result documents but applies htmlspecialchars on all fields retrieved
+	 * from solr except the configured fields in
+	 *
+	 * plugin.tx_solr.search.skipEscapeFields
+     *
+     * @return array
+	 */
+	public function getResultDocumentsEscaped()
```

Comment:
For two reasons:

1. The behaviour is changing when we change it directrly. Therefore i've added a new method and marked the new as deprecated and update all usages.

2. Also a boolean parameter in the signature would be bad style: ( http://programmers.stackexchange.com/questions/147977/is-it-wrong-to-use-a-boolean-parameter-to-determine-behavior) or over time hard to maintain

Therefore i think this was the best choice